src/spyglass/spikesorting/utils_burst.py

```diff
@@ -0,0 +1,264 @@
+from typing import Dict, List, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+from datajoint.expression import QueryExpression
+
+from spyglass.utils import logger
+
+
+def validate_pairs(
+    query: QueryExpression, pairs: Union[Tuple, List[Tuple]]
+) -> List[Tuple]:
+    """Validate unit pairs for a given query.
+
+    Checks that the passed pairs are valid pairs in the query. May flip the
+    order of the pairs if the reverse pair is found. Query is assumed to have
+    columns unit1, unit2.
+    """
+
+    def validate_pair(
+        query: QueryExpression, unit1: int, unit2: int
+    ) -> Tuple[int, int]:
+        """Ensure that unit1, unit2 is a valid pair in the table."""
+        template = "unit1={unit1} AND unit2={unit2}"
+        if query & template.format(unit1=unit1, unit2=unit2):
+            return unit1, unit2
+        elif query & template.format(unit1=unit2, unit2=unit1):
+            logger.warning(f"Using reverse pair {unit1}, {unit2}")
+            return unit2, unit1
+        else:
+            logger.warning(f"No entry found for pair {unit1}, {unit2}")
+            return None
+
+    if isinstance(pairs, tuple) and len(pairs) == 2:
+        pairs = [pairs]
+
+    valid_pairs = []
+    for p in pairs:
+        if valid_pair := validate_pair(query, *p):
+            valid_pairs.append(valid_pair)
+
+    if not valid_pairs:
+        raise ValueError("No valid pairs found")
+    return valid_pairs
+
+
+def plot_burst_xcorrel(fig, pairs, ccgs_e, bins):
```

Comment:
I don't understand why `fig` is passed here and then overwritten by the fig in `plt.subplots`. Is it just to keep consistency?

Reply:
Just because that's how it was previously written - I trusted that a passed arg was being updated. I'll remove